docs/proposals/organizations.md

```diff
@@ -0,0 +1,125 @@
+
+This approach is relatively easy to build, but it is limited because it doesn't reserve quota for a given project.  A more complicated approach would be to allow killing of "extra" resources and then provide a rate and ceiling like tc.  This would have an overall cap and allow "borrowing" of quota when resources are plentiful, but then selectively kill resources when resources are scarce to allow for project guarantees.  With enough effort, this sort of solution could be layered on top of kube by setting the ceiling for the existing quota controller and having a separate resource for the guarantee and a separate controller for the killer.
+
+I'm sure there will be complaints no matter what we do.
+
+
+## Default Policy
+org-admins should have power somewhere between a cluster-admin and a project-admin.  Since the idea of an Organization doesn't scope cleanly, we can make use of the `attributeRestrictions` to run custom logic for a cluster-scoped rule.  We can create a `ClusterRoleBinding` to `system:authenticated` that allows a `ClusterRole` with the correct access to every project that the is owned by a Organization that the user is an org-admin for.  That makes it impossible for an org-admin to accidentally (or intentionally) remove his powers on a project.
+
+
+## Project ownership
+Projects will have an annotation that indicates which Organization owns them.  If we don't allow modifications to project by default, then a cluster-admin can sort it out for now until we make an offer/request/approval flow.
+
+
+## Org Member powers
+A given user can:
+ 1.  See which orgs he's a part of.  This will require a reverse-lookup map/cache and probably a subresource on user.
+ 2.  Leave an organization that he's tied to by MemberUser, but not one that he is tied to by MemberGroup.  That's an odd wrinkle.
+
+
+## OrgGroups Overview
+Project administrators need to be able to define groups so they can efficiently manage policy on their projects, however a project administrator is unlikely to be allowed to manage cluster-scoped group membership across all projects.  A single project administrator (or set of project administrators) is likely to own multiple related projects (think single department running multiple projects).  To allow cluster-scoped group definitions to be shared across multiple related projects, we'll introduce the concept of OrgGroups (scoped sets of users) and Organizations (things that reference Users and own OrgGroups).
+
+Organizations are cluster-scoped resources that contains a set of OrgGroups, a list of admins (users that can manage the groups), and a list of members (users that can be part of the OrgGroups).  Groups are cluster-scoped sets of users that are managed by a cluster-admin.  OrgGroups are organization scoped sets of users that are managed by the Organization.AdminUsers.  Projects may have at most one owning Organization.  OrgGroups may only be referenced in RoleBindings of Projects owned by the Organization.
```

Comment:
> Is an Organization referencing its OrgGroups? How?

> contains a set of OrgGroups

So OrgGroups are a part of the Organization object? I don't see it in the definition above.